### ISSUE TITLE
Enough Plumbers: Cannot start game

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#10375]
### Describe the bug

Link to SWF: http://i.notdoppler.com/files/enoughplumbers.swf

Clicking the Play button in [Enough Plumbers](http://www.notdoppler.com/enoughplumbers.php) does not start the game. The error is:
```
ERROR ruffle_core::display_object::movie_clip: Error occured when running AVM2 frame script: TypeError: Type Coercion failed: cannot convert Object(StageObject(StageObject { name: RuntimeFlash, ptr: 0x1f0640fba30 })) to mx.core.IFlexDisplayObject.
        at mx.managers::SystemManager/initializeTopLevelWindow()
        at mx.managers::SystemManager/http://www.adobe.com/2006/flex/mx/internal::docFrameHandler()
        at flash.display::MovieClip/flash::display::MovieClip::nextFrame()
        at mx.managers::SystemManager/deferredNextFrame()
        at mx.managers::SystemManager/preloader_initProgressHandler()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at mx.preloaders::Preloader/timerHandler()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at flash.utils::Timer/onUpdate()
```

### Expected behavior

The game should start.

### Affected platform

Desktop app

### Operating system

Windows 10

**Comment by Bale001** [comment]
This is caused by our improper interface inheritance support

**Comment by AllinolCP** [comment]
[minimal-repro.zip](https://github.com/ruffle-rs/ruffle/files/11217986/minimal-repro.zip)
Minimal reproduction of the issue

**Comment by n0samu** [comment]
Fixed by #11150
